src/lib/fragments/AgGrid.react.js

```diff
@@ -1023,23 +1023,23 @@ export default class DashAgGrid extends Component {
         if (!gridApi) {
             return;
         }
+        const rowPosition = scrollTo.rowPosition ? 'top' : scrollTo.rowPosition;
         if (scrollTo.rowIndex || scrollTo.rowIndex === 0) {
-            const rowPosition =
-                scrollTo.rowPosition === null ? 'top' : scrollTo.rowPosition;
             gridApi.ensureIndexVisible(scrollTo.rowIndex, rowPosition);
-        }
-        if (scrollTo.rowId) {
-            const rowPosition =
-                scrollTo.rowPosition === null ? 'top' : scrollTo.rowPosition;
-
+        } else if (scrollTo.rowId) {
             const node = gridApi.getRowNode(scrollTo.rowId);
             gridApi.ensureNodeVisible(node, rowPosition);
+        } else if (scrollTo.data) {
+            gridApi.forEachNode((node) => {
+                if (includes(node.data, [scrollTo.data])) {
+                    gridApi.ensureNodeVisible(node, rowPosition);
```

Comment:
find some way to break it once it finds the first match, this way you dont have to keep going through the whole dataset

Reply:
@BSd3v you resolved this but I think we can do better, even if `forEachNode` doesn't have a way to short-circuit:
(1) outside the loop `let scrolled = false;`, then inside the loop `if (!scrolled && equals(node.data, scrollTo.data)) { scrolled = true; ... }`
(2) instead of `forEachNode` we should use I think `forEachNodeAfterFilterAndSort` because there's no point scrolling to a hidden node.